### DISCUSSION TITLE
I need help understanding Centered-log ratio transformation calculations.

### DISCUSSION THREAD

**Discussion by SlowSD** [satijalab/seurat#9187] [Q&A]
I have this example dataset.
```
      Cell1 Cell2 Cell3 Cell4 Cell5
GeneA     2    10     9     8     9
GeneB     2     4    10     8     6
GeneC     9     3     4     8     9
GeneD     1     5     2     2     8
GeneE     5     8    10     7     2
```

Upon using `NormalizeData(seurat_object, normalization.method = "CLR")`, the normalization takes place across features and the following dataset is obtained.

```
          Cell1     Cell2     Cell3     Cell4     Cell5
GeneA 0.2271068 0.8219007 0.7642287 0.7030260 0.7642287
GeneB 0.2734809 0.4880058 0.9449407 0.8145365 0.6645402
GeneC 0.8189869 0.3525801 0.4470195 0.7548487 0.8189869
GeneD 0.2252369 0.8167343 0.4089516 0.4089516 1.1055731
GeneE 0.5528466 0.7798341 0.9068010 0.7097444 0.2587250
```

**Could anyone please help me understand the mathematical calculations behind the CLR normalization. I have tried dividing feature counts by geometric mean and taking the natural log. However, I believe that I obtained totally different entries.**

Here is the screenshot of the procedure.
![image](https://github.com/user-attachments/assets/66d25d91-1c95-47eb-8f2b-034b96aab0b7)

Any help would be appreciated.

Regards.

**Comment by Truongphi20** [comment]
@SlowSD This is my A-code (Algorithm code capturing the core logic without auxiliary) of CLR transform, hope it could help you.

[clr_normalise.R.txt](https://github.com/user-attachments/files/27903446/clr_normalise.R.txt)
